docs(guides/data-loading | api/conventions): fix type inference & add json import (#4609)
remix-run/remix: docs/guides/data-loading.md

If you are using TypeScript, you can use type inference to use Prisma Client generated types on when calling `useLoaderData`. This allows better type safety and intellisense when writing your code that uses the loaded data.

```tsx filename=tsx filename=app/routes/products/$productId.tsx

> import type { LoaderArgs } from '@remix-run/node'; // or cloudflare/deno

Reviewer: ```suggestion
import type { LoaderArgs } from "@remix-run/node"; // or cloudflare/deno
```

  return product;
}

export const loader = async ({ params }: LoaderArgs) => {

>  const product = await getLoaderData(params.productId)

Reviewer: ```suggestion
  const product = await getLoaderData(params.productId);
```

🤓 🙈

Author: 🙈

>  const product = await getLoaderData(params.productId);

Reviewer: ```suggestion
  const product = await getLoaderData(params.productId);
```

  return product;
}

export const loader = async ({ params }: LoaderArgs) => {
  let LoaderData: Awaited<ReturnType<typeof getLoaderData>> | null = null
  LoaderData = await getLoaderData(`${params.productId}`)

>   return json({product: LoaderData});

Reviewer: is this all needed? could you try:

```ts
export const loader = async ({ params }: LoaderArgs) => {
  const product = await getLoaderData(params.productId);

  return json({  product });
```

and see if type inference works?

Author: yea thanks it's better. type inference work.
